src/core/plugin/luapi_application.h

```diff
@@ -1976,6 +1977,135 @@ static int applib_openFile(lua_State* L) {
+ */
+static int applib_addImagesFromFilepath(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+
+    int cntParams = lua_gettop(L);
+
+    for (int iParam{1}; iParam <= cntParams; iParam++) {
+        luaL_checktype(L, iParam, LUA_TTABLE);
+
+        lua_getfield(L, iParam, "path");
+        lua_getfield(L, iParam, "x");
+        lua_getfield(L, iParam, "y");
+        lua_getfield(L, iParam, "width");
+        lua_getfield(L, iParam, "height");
+        lua_getfield(L, iParam, "scale");
+        lua_getfield(L, iParam, "aspectRatio");
+
+        // stack now has the following:
+        //    1  = current params table
+        //    ... potentially other params tables
+        //   -7 = filepath
+        //   -6 = x coordinate
+        //   -5 = y coordinate
+        //   -4 = width (in pixel)
+        //   -3 = height (in pixel)
+        //   -2 = scale
+        //   -1 = aspectRatio
+
+        // fetch the parameters and check for validity. If parameter is invalid -> hard error
+        double x = luaL_optnumber(L, -6, 0);
+        double y = luaL_optnumber(L, -5, 0);
+
+        int heightParam = luaL_optinteger(L, -3, -1);
+        if (heightParam <= 0 && heightParam != -1)
+            return luaL_error(L, "Invalid height given, must be positive integer or -1 to deactivate manual setting.");
+
+        int widthParam = luaL_optinteger(L, -4, -1);
+        if (widthParam <= 0 && widthParam != -1)
+            return luaL_error(L, "Invalid width given, must be positive integer or -1 to deactivate manual setting.");
+
+        double scale = luaL_optnumber(L, -2, 1);
+        if (scale <= 0)
+            return luaL_error(L, "Invalid scale given, must be a positive number.");
+
+        bool aspectRatio{true};
+        if (lua_isnil(L, -1)) {
+            // map nil to true (not typical for lua) so that the default value is true
+            aspectRatio = true;
+        } else {
+            // use the typical lua version of booleans (everything different than false (any set value) is true)
+            aspectRatio = lua_toboolean(L, -1);
+        }
+
+        const char* path = luaL_checkstring(L, -7);
+        if (!path)
+            return luaL_error(L, "no 'path' parameter was provided.");
+
+        xoj::util::GObjectSPtr<GFile> file(g_file_new_for_path(path), xoj::util::adopt);
+        if (!g_file_query_exists(file.get(), NULL)) {
+            lua_pushstring(L, "Error: file does not exist.");  // soft error
+            return 1;
+        }
+
+        XojPageView* pv = control->getWindow()->getXournal()->getViewFor(control->getCurrentPageNo());
+        ImageHandler imgHandler(control, pv);
+        auto [succ, img, width, height] = imgHandler.createImage(file.get(), x, y);
+        if (!succ) {
+            lua_pushstring(L, "Error: creating the image failed.");  // soft error
```

Comment:
Here and elsewhere, the user should be informed about which image produced the failure.

Reply:
Alright. `pushstring` does not support a format string so we'd need something like `sprintf`. ~Generally I'd use more (modern cpp formatting with `format`](https://en.cppreference.com/w/cpp/utility/format)~ (only available as of Cpp20, we're using Cpp17).
So we should use [`cstdio`](https://en.cppreference.com/w/cpp/io/c/fprintf), to format a string like `Error: creating the imageNo {indexOfTheImage} failed.` right? (or have you thought of another way of identifying the image? Maybe we could also simply return a value per passed image encoding errors might get confusing as we shouldn't use `nil` for success as this makes iterating over the returned value difficult in lua)
As we're working with lua always the question arises should the index start with `1` like it's common in lua or with `0` common in most programming.

Reply:
I actually thought about adding the path (not the index) to the string. There is `lua_pushfstring` for formatting.

Currently the "soft error" while returning early (before the other images have been processed) doesn't make a lot of sense. It would be better to accumulate these messages and return them at the end or otherwise use a "hard error".

Yes, in Lua the index should start with 1.

Reply:
`lua_pushfstring` oh thanks :+1: 

I'd keep the "soft error" because a file not being present might be pretty common and is quite like the `ressource not available` when we were discussing about "soft/hard errors".

So I think we have two options:
1. return the errors like `return succ, succ, "Error: xyz", succ` (lua code)
2. return them like `return {succ, succ, "Error: xyz", succ}`. In this case we might even be able to eliminate the `true` elements (though we must note that `ipairs` will most probably not applicable anymore then).

I'm not decided yet what to favor, 1. looks a bit more lua idiomatic but 2. has its advantages as well. In the end, we must be clear that with `table.unpack` and `table.pack` these options are not that different.

And we need to decide what `succ` should be. The value common in lua would be `nil`. The drawback of this value is that it breaks `ipairs` in arraylike tables. The value nearest would be `false` but I'm not sure if this value is not a bit confusing at this point.

Reply:
Alright, I decided to go with 1. for now using `succ = nil`. If the user wants a table (which can't be iterated with `ipairs` because of the `nil`), this has to be created in the lua code with `table.pack` or by wrapping with `{app.addImagesFromFilepaths()}`.
See 12e7c8b29